File: src/main/java/org/micoli/php/symfony/list/configuration/RoutesConfiguration.java
package org.micoli.php.symfony.list.configuration;

import io.swagger.v3.oas.annotations.media.Schema;
import org.micoli.php.configuration.models.DisactivableConfiguration;

public final class RoutesConfiguration implements DisactivableConfiguration {
    @Override
    public boolean isDisabled() {
        return !enabled;
    }

    @Schema(description = "Enabler for panel of routes")
    public boolean enabled = true;

    @Schema(description = "List of namespaces where routes are searched")
    public String[] namespaces = {"\\App", "\\Application"};

    @Schema(description = "Attribute used to detect routes")
    public String attributeFQCN = "\\Symfony\\Component\\Routing\\Attribute\\Route";
}

File: src/main/kotlin/org/micoli/php/symfony/list/configuration/RoutesConfiguration.kt
package org.micoli.php.symfony.list.configuration

import io.swagger.v3.oas.annotations.media.Schema
import org.micoli.php.configuration.models.DisactivableConfiguration

class RoutesConfiguration : DisactivableConfiguration {
    override fun isDisabled(): Boolean {
        return !enabled
    }

    @Schema(description = "Enabler for panel of routes") var enabled: Boolean = true

    @Schema(description = "List of namespaces where routes are searched")
    var namespaces: Array<String> = arrayOf("\\App", "\\Application")

    @Schema(description = "Attribute used to detect routes")
    var attributeFQCN: String = "\\Symfony\\Component\\Routing\\Attribute\\Route"
}
